arrows/ceres/register_algorithms.cxx

```diff
@@ -33,26 +33,39 @@
  * \brief Ceres algorithm registration implementation
  */
 
-#include "register_algorithms.h"
+#include <arrows/ceres/kwiver_algo_ceres_export.h>
+#include <vital/algo/algorithm_factory.h>
 
-
-#include <arrows/algorithm_plugin_interface_macros.h>
 #include <arrows/ceres/bundle_adjust.h>
 
 
 namespace kwiver {
 namespace arrows {
 namespace ceres {
 
-/// Register Ceres algorithm implementations with the given or global registrar
-int register_algorithms( vital::registrar &reg )
+extern "C"
+KWIVER_ALGO_CERES_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  REGISTRATION_INIT( reg );
+  static auto const module_name = std::string( "algorithms.ceres" );
+  if (vpm.is_module_loaded( module_name ) )
```

Comment:
This model was used by sprokit and it keeps the first module loaded. If a local module is to override another one, it may be easier to force the desired version to be loaded first rather than last.  (IIRC, the KWIVER_PLUGIN_PATH is used before the other ones.)

Registering and checking a module name is not required, but the plugin manager checks to see if a plugin_factory is already registered and will throw an exception if a factory is already registered.